packages/react-core/src/components/Card/CardHeader.tsx

```diff
@@ -1,21 +1,53 @@
 import * as React from 'react';
 import { css } from '@patternfly/react-styles';
 import styles from '@patternfly/react-styles/css/components/Card/card';
+import { CardContext } from './Card';
+import { Button } from '../Button';
+import AngleRightIcon from '@patternfly/react-icons/dist/js/icons/angle-right-icon';
 
 export interface CardHeaderProps extends React.HTMLProps<HTMLDivElement> {
   /** Content rendered inside the CardHeader */
   children?: React.ReactNode;
   /** Additional classes added to the CardHeader */
   className?: string;
+  /** ID of the card header */
+  id?: string;
+  /** Callback expandable card */
+  onExpand?: (cardId: string) => void;
+  /** Additional props for expandable toggle button */
+  toggleButtonProps?: any;
 }
 
 export const CardHeader: React.FunctionComponent<CardHeaderProps> = ({
   children = null,
   className = '',
+  id = '',
+  onExpand,
+  toggleButtonProps,
   ...props
 }: CardHeaderProps) => (
-  <div className={css(styles.cardHeader, className)} {...props}>
-    {children}
-  </div>
+  <CardContext.Consumer>
+    {({ cardId }) => (
+      <div className={css(styles.cardHeader, className)} id={id} {...props}>
+        {onExpand && (
+          <div className={css(styles.cardHeaderToggle)}>
+            <Button
+              variant="plain"
+              type="button"
+              onClick={() => {
+                onExpand(cardId);
```

Comment:
should pass the event along as well, and I think typically event should be the first parameter passed back?